ballerina/modules/soap11/soap11.bal

```diff
@@ -81,13 +82,13 @@ public isolated client class Client {
                 response = check soap:sendReceive(securedBody, self.soapClient, action, headers, path, false);
             }
             lock {
-                soap:OutboundSecurityConfig? outboundSecurity = self.outboundSecurity.clone();
+                soap:InboundSecurityConfig? inboundSecurity = self.inboundSecurity.clone();
                 do {
-                    if outboundSecurity is soap:OutboundSecurityConfig && outboundSecurity != {} {
+                    if inboundSecurity is soap:InboundSecurityConfig && inboundSecurity != {} {
                         if response is xml {
-                            return check soap:applyOutboundConfig(outboundSecurity.clone(), response.clone(), false);
+                            return check soap:applyOutboundConfig(<wssec:InboundConfig>inboundSecurity.clone(), response.clone(), false);
                         } else {
-                            return check soap:applyOutboundConfig(outboundSecurity.clone(), 
+                            return check soap:applyOutboundConfig(<wssec:InboundConfig>inboundSecurity.clone(), 
```

Comment:
it seems we can avoid these explicit castings if we add a type check in L87, isn't it?